test/graphql/create_project_mutation_test.rb

```diff
@@ -0,0 +1,267 @@
+    assert_equal 'New Project One Description', data['project']['description']
+
+    project = Project.last
+    assert_equal group.id, project.parent.id
+    assert_equal 'new-project-one', project.path
+    assert_equal 'group-1/new-project-one', project.full_path
+  end
+
+  test 'createProject mutation should work with valid params, group puid, and api scope token' do
+    group = groups(:group_one)
+
+    result = IridaSchema.execute(CREATE_PROJECT_USING_GROUP_PUID_MUTATION,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { groupPuid: group.puid,
+                                              name: 'New Project One',
+                                              description: 'New Project One Description' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createProject']
+
+    assert_not_empty data, 'createProject should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['project']
+
+    assert_equal 'New Project One', data['project']['name']
+    assert_equal 'New Project One Description', data['project']['description']
+
+    project = Project.last
+    assert_equal group.puid, project.parent.puid
+    assert_equal 'new-project-one', project.path
+    assert_equal 'group-1/new-project-one', project.full_path
+  end
+
+  test 'createProject mutation should work with valid params, no group id/puid, and api scope token' do
+    result = IridaSchema.execute(CREATE_PROJECT_MUTATION,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Project One',
+                                              description: 'New Project One Description' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createProject']
+
+    assert_not_empty data, 'createProject should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['project']
+
+    assert_equal 'New Project One', data['project']['name']
+    assert_equal 'New Project One Description', data['project']['description']
+
+    project = Project.last
+    assert_equal @user.namespace.puid, project.parent.puid
+    assert_equal 'new-project-one', project.path
+    assert_equal 'john.doe_at_localhost/new-project-one', project.full_path
+  end
+
+  test 'createProject mutation should work with custom path' do
+    result = IridaSchema.execute(CREATE_PROJECT_MUTATION_WITH_CUSTOM_PATH,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Project One',
+                                              description: 'New Project One Description',
+                                              path: 'New Custom Path' })
```

Comment:
**nitpick (blocking):** With comment above this test should raise an error due to the path not passing validation.

Reply:
Added new tests for passing/failing with custom path in https://github.com/phac-nml/irida-next/pull/795/commits/de833800d4766f5f65ef80f657f56cb00bc8505c